Assembler.Base/FramesAssembler.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Microsoft.Extensions.Logging;
+
+namespace Assembler.Base
+{
+    public class FramesAssembler<TFrame> : IAssembler<TFrame>
+        where TFrame : BaseFrame
+    {
+        private readonly IResolver<AssemblingPosition, IFrameHandler<TFrame>> _typeToHandlerResolver;
+        private readonly ILogger _logger;
+
+        public FramesAssembler(IResolver<AssemblingPosition, IFrameHandler<TFrame>> typeToHandlerResolver,
+            ILoggerFactory loggerFactory)
+        {
+            _typeToHandlerResolver = typeToHandlerResolver;
+            _logger = loggerFactory.CreateLogger(GetType());
+        }
+
+        public void Assemble(TFrame frame)
+        {
+            IFrameHandler<TFrame> handler = ResolveHandler(frame);
+
+            _logger.LogDebug($"Frame [{frame.Guid}] is being sent to the handler.");
+
+            HandleFrame(frame, handler);
+        }
+
+        private IFrameHandler<TFrame> ResolveHandler(TFrame frame)
+        {
+            try
+            {
+                return _typeToHandlerResolver.Resolve(frame.AssemblingPosition);
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"No matching resolver was found for the message [{frame.Guid}]," +
+                                   $"It won't be used in the assembling process. \n {e}");
+                return null;
+            }
+        }
+
+        private void HandleFrame(TFrame frame, IFrameHandler<TFrame> handler)
+        {
+            try
+            {
+                handler?.Handle(frame);
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"Failed handling the message [{frame.Guid}]," +
```

Comment:
Error instead of Warn

Reply:
It should be the sames as the previous one

Reply:
No it shouldnt, because if you got a frame which isnt configured thats ok, its a reasonable flow (in example, in case you parsed it but still doesnt handle it), but there is no good and pre-known reason for the handle to throw exception. It must be Error

Reply:
f